sql/core/src/test/java/test/org/apache/spark/sql/JavaBeanDeserializationSuite.java

```diff
@@ -115,6 +119,74 @@ public void testBeanWithMapFieldsDeserialization() {
     Assert.assertEquals(records, MAP_RECORDS);
   }
 
+  @Test
+  public void testSpark22000() {
+    List<Row> inputRows = new ArrayList<>();
```

Comment:
Could you add more tests for upcast failure cases?

Reply:
Honestly I'm not sure which cases other that this fail without upcast. Please share if you have any, and I'll add them.

Reply:
How about this case?
```

scala> :paste
class BeanClass extends Serializable {
  private var id: Int = _
  def getId(): Int = id
  def setId(i: Int): Unit = { id = i }
}
implicit val encoder = org.apache.spark.sql.Encoders.bean(classOf[BeanClass])
Seq("1", "2", "a").toDF("id").as[BeanClass].collect

org.apache.spark.sql.AnalysisException: Cannot up cast `id` from string to int as it may truncate
The type path of the target object is:
- field (class: "int",
name: "id")
- root class: "BeanClass"
You can either add an explicit cast to the input data or choose a higher precision type of the field in the target object;
  at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveUpCast$.org$apache$spark$sql$catalyst$analysis$Analyzer$ResolveUpCast$$fail(Analyzer.scala:2536)
  at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveUpCast$$anonfun$apply$29$$anonfun$applyOrElse$129.applyOrElse(Analyzer.scala:2552)
  at org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveUpCast$$anonfun$apply$29$$anonfun$applyOrElse$129.applyOrElse(Analyzer.scala:2547)
  at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$transformDown$1(TreeNode.scala:258)
  at org.apache.spark.sql.catalyst.trees.CurrentOrigin$.withOrigin(TreeNode.scala:72)
  at org.apache.spark.sql.catalyst.trees.TreeNode.transformDown(TreeNode.scala:258)
  at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$transformDown$3(TreeNode.scala:263)
  at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$mapChildren$1(TreeNode.scala:328)
  at org.apache.spark.sql.catalyst.trees.TreeNode.mapProductIterator(TreeNode.scala:189)
  at org.apache.spark.sql.catalyst.trees.TreeNode.mapChildren(TreeNode.scala:326)
  at org.apache.spark.sql.catalyst.trees.TreeNode.transformDown(TreeNode.scala:263)
  at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$transformDown$3(TreeNode.scala:263)
  at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$mapChildren$2(TreeNode.scala:345)
```

Reply:
Ah OK I misread your comment. Looks like you would like to have another test which can't be mitigated with upcast so the query is failing even with upcast. I'll add it. Thanks!

Reply:
Thanks!

Reply:
Addressed.